### ISSUE TITLE
[3.7-beta.2] Changes to matrix blocks are incorrectly merged into drafts

### ISSUE THREAD

**Issue by wsydney76** [craftcms/cms#8092]
### Description

When working in parallel on the same entry, e.g. on the current revision (user A) and a provisional draft (user B), changes to matrix blocks made in the current revision are not merged correctly into the draft.

Didn't test other scenarios, because i think regardless how clever the merge algorithm is, it will likely result in unwanted changes. Will add a FR to opt out.

### Steps to reproduce

1. Create and save an entry with one matrix block (user A)
![image](https://user-images.githubusercontent.com/60010597/122726601-75d80f80-d276-11eb-98e5-8412fdb91a21.png)

2. Add one matrix block (user B), not saved
![image](https://user-images.githubusercontent.com/60010597/122726858-b46dca00-d276-11eb-8657-bd5403ce220b.png)

3. Add on matrix block (user A) and save
![image](https://user-images.githubusercontent.com/60010597/122727138-0282cd80-d277-11eb-81f3-5e473fa1cf2e.png)

4. Reload edit screen for user B, changes are merged, resulting in duplicate blocks
![image](https://user-images.githubusercontent.com/60010597/122727410-47a6ff80-d277-11eb-94d6-157d70cbd080.png)

The only visible feed back is a yellow bar marking the whole matrix field, not just the merged changes.

### Additional info

- Craft version:
- PHP version:
- Database driver & version:
- Plugins & versions:

**Comment by brandonkelly** [comment]
Guessing this has something to do with site propagation… can you tell me:

- How many sites is this section enabled for?
- What is the section’s Propagation Method set to?
- What is the Matrix field’s Propagation Method set to?

**Comment by wsydney76** [comment]
@brandonkelly No, this is a fresh install with just one site.

If I follow the same steps as above in a multi site installation and translate each block immediatly after adding it, i get the same result (duplicate blocks) for each site.

Set up to 'Save entries to all sites enabled...' /  'Save blocks to all sites the owner....'

**Comment by brandonkelly** [comment]
Thanks! Fixed for the next release.

**Comment by brandonkelly** [comment]
3.7.0-beta.3 is out now with the fix.

**Comment by rob-baker-ar** [comment]
If anyone else sees this, for me, it was caused by more than one provisional draft existing per entry per user - there should only be one. The effect of this was that Craft would merge those drafts into the one being viewed making it look like matrix blocks were being duplicated. They were not - just getting erroneously merged in.

Anyway, craft `3.7.9` has a command added that ensures there is only a single provisional draft per entry per user. While it did not fix whatever caused the situation my database was in _(still not sure what did cause it!)_, it did prevent the issue getting worse and so far, I have not seen it again.

```bash
php craft utils/prune-provisional-drafts
```
